parcels/field.py

```diff
@@ -156,12 +156,15 @@ def from_netcdf(cls, filenames, variable, dimensions, indices=None, grid=None,
         indices = {} if indices is None else indices.copy()
         with NetcdfFileBuffer(dimension_filename, dimensions, indices) as filebuffer:
             lon, lat = filebuffer.read_lonlat
-            depth = filebuffer.read_depth
             indices = filebuffer.indices
             # Check if parcels_mesh has been explicitly set in file
             if 'parcels_mesh' in filebuffer.dataset.attrs:
                 mesh = filebuffer.dataset.attrs['parcels_mesh']
 
+        dimension_filename = filenames[0]
+        with NetcdfFileBuffer(dimension_filename, dimensions, indices) as filebuffer:
+            depth = filebuffer.read_depth
```

Comment:
This isn't a very pretty solution, I'd say. This only works for nemo, but may leave users of other datasets out. Even if they would put `depth` in their `dimension_filename`, that would be overloaded here. Should we implement this in a way that is more flexible for different use cases, not only nemo?

Reply:
Why don't we replace the `mesh_mask` file which is not super explicit by a dimesion_file, which can be a file (the mesh_mask) or a dict of files, per dimension.
This follows the logic as everywhere else, no?

Reply:
yes, that is a good idea. Although we need to support also that `dimensions` can also be a dictionary of dictionaries, for each variable. So then we could have two dictionaries: `dimension_names` and `dimension_files`. 

The former is what we now have as `dimensions`, and the latter would default to `filenames[0]` (i.e. assume that all dimensions information is simply in the variable files) but can be overloaded as either a single dictionary (`dimension_files['lon']) or a dictionary-of-dictionaries (`dimension_files['U']['lon']). 
Makes sense?